nilearn/image/image.py

```diff
@@ -1284,6 +1284,11 @@ def clean_img(
         imgs_ = new_img_like(
             imgs_, data.T.reshape(imgs_.shape), copy_header=True
         )
+    if "sample_mask" in clean_kwargs:
+        sample_shape = imgs_.shape[:3] + clean_kwargs["sample_mask"].shape
+        imgs_ = new_img_like(
+            imgs_, data.T.reshape(sample_shape), copy_header=True
+        )
```

Comment:
I'm not sure at having this at the end like this as it overwrites results from the if else statement above. Don't we need to make sure that whether "sample_mask" is given or not, if mask_img is not None that the cleaned signal is first unmasked?

Reply:
I changed it from an elif statement because they are not mutually exclusive, but this does seem to overwrite the mask_img portion. We could move this if statement to be above the mask_img if statement, or we can somehow incorporate the changes made from the previous statement into this one so that the shape is changed with the sample_mask but does not overwrite the changes from the mask_img.

Reply:
So what is stored in the data variable does drop the samples already when returned from signal.clean so if both sample_mask and mask_img are given the unmasking function will correctly reshape and put the data into a nifti object. I would put this new code back into an `elif` statement above the last `else` statement.